flink-connectors/flink-jdbc/src/test/java/org/apache/flink/api/java/io/jdbc/JDBCTableSourceITCase.java

```diff
@@ -153,4 +153,34 @@ public void testProjectableJDBCSource() throws Exception {
 				"2020-01-01T15:36:01.123456,101.1234");
 		StreamITCase.compareWithList(expected);
 	}
+
+	@Test
+	public void testScanQueryJDBCSource() throws Exception {
+		StreamExecutionEnvironment env = StreamExecutionEnvironment.getExecutionEnvironment();
+		EnvironmentSettings envSettings = EnvironmentSettings.newInstance()
+			.useBlinkPlanner()
+			.inStreamingMode()
+			.build();
+		StreamTableEnvironment tEnv = StreamTableEnvironment.create(env, envSettings);
+
+		final String testQuery = "SELECT id FROM " + INPUT_TABLE;
+		tEnv.sqlUpdate(
+			"CREATE TABLE test(" +
+				"id BIGINT" +
+				") WITH (" +
+				"  'connector.type'='jdbc'," +
+				"  'connector.url'='" + DB_URL + "'," +
+				"  'connector.table'='whatever'," +
+				"  'connector.read.query'='" + testQuery + "'" +
+				")"
+		);
+
+		StreamITCase.clear();
```

Comment:
I think using the static `StreamITCase` functionality here is an anti-pattern that we should avoid. (I know that it's widely used in the code base, but we should rid of it at some point.)

Ideally, with the FLIP-84 work it should be as easy as
```
TableResult tableResult = tEnv.executeSql("SELECT timestamp6_col, decimal_col FROM " + INPUT_TABLE);
tableResult.collect();
```

but that is not yet in master. I'm inclined to wait for a but but basically have this PR ready in a mergable state.

Reply:
I just copied the code from another test and I modifier it. I didn't ask myself too much why there's the need to call StreamITCase.clear()

Reply:
I understand, the code code that is there is already not good. 😞

Reply:
Maybe we can merge this and open an issue to fix the tests in this class to the right way when FLIP-84 will be ready